File: darknight/dashboard/src/api/portal/invite.ts
import { http } from '@/config/axios'

export interface InviteSummary {
  balance: number
  currency: string
  registered_count: number
  commission_rate: number
  pending_commission: number
  total_commission: number
}

export interface InviteCode {
  code: string
  created_at: string
  invite_url: string
}

export interface InvitePayout {
  paid_at: string
  amount: number
  currency: string
}

export function fetchInviteSummary() {
  return http<InviteSummary>('/invite/summary')
}

export function fetchInviteCodes() {
  return http<InviteCode[]>('/invite/codes')
}

export function generateInviteCode() {
  return http<InviteCode>('/invite/codes', { method: 'POST' })
}

export function fetchInvitePayouts() {
  return http<InvitePayout[]>('/invite/payouts')
}

export function formatInviteTime(iso: string): string {
  const date = new Date(iso)
  const pad = (n: number) => String(n).padStart(2, '0')
  return `${date.getFullYear()}/${pad(date.getMonth() + 1)}/${pad(date.getDate())} ${pad(date.getHours())}:${pad(date.getMinutes())}`
}

File: darknight/models/invite.py
from datetime import datetime

from pydantic import BaseModel


class InviteSummaryResponse(BaseModel):
    balance: float = 0
    currency: str = "USD"
    registered_count: int = 0
    commission_rate: float = 0.10
    pending_commission: float = 0
    total_commission: float = 0


class InviteCodeResponse(BaseModel):
    code: str
    created_at: datetime
    invite_url: str


class InvitePayoutResponse(BaseModel):
    paid_at: datetime
    amount: float
    currency: str = "USD"
